iceberg/src/main/scala/com/nvidia/spark/rapids/iceberg/parquet/GpuCoalescingIcebergParquetReader.scala

```diff
@@ -101,7 +101,7 @@ class GpuCoalescingIcebergParquetReader(val files: Seq[IcebergPartitionedFile],
         extraInfo
           .asInstanceOf[IcebergParquetExtraInfo]
           .postProcessor
-          .process(batch)
+          .process(GpuColumnVector.incRefCounts(batch))
```

Comment:
This feels odd to me. Is someone else going to close this batch too? I also cannot find where this is called, or the definition o this in the parent class, so I ma not sure how to understand the ownership of batch or how it is intended to be used.

Reply:
But that is exactly why it is a problem.

The post processing does a retry with no splits.

https://github.com/NVIDIA/spark-rapids/blob/22d34cb6c5d8076eaa7af8649107785a77fb91c9/iceberg/src/main/scala/com/nvidia/spark/rapids/iceberg/parquet/GpuParquetReaderPostProcessor.scala#L71

If we inc the ref counts then not all of the data is spillable when we enter the retry block.